src/appliance-onboarding-script/appliance_setup/pkgs/_appliance_setup.py

```diff
@@ -203,14 +207,19 @@ def _deploy_and_create_appliance(self) -> str:
         with TempChangeDir(self._temp_dir):
             config = self._config
             apiserver_address = config['applianceControlPlaneIpAddress']
-            if is_api_server_reachable and not confirm_prompt(f'An ApiServer is already reachable on endpoint {apiserver_address}. Deployment will be skipped. Do you want to continue?'):
+            # Removing confirm_prompts for automated testing 
+            # TODO Check what needs to be done in case there is a reachable ApiServer
+            # For now, we consider the Api server to be correctly configured and don't deploy a new one.
+            if is_api_server_reachable and (self._isAutomated or not confirm_prompt(f'An ApiServer is already reachable on endpoint {apiserver_address}. Deployment will be skipped. Do you want to continue?')):
                 raise ProgramExit('User chose to exit the program.')
             if not is_api_server_reachable:
                 logging.info('Deploying appliance...')
                 res, err = az_cli('arcappliance', 'deploy', 'vmware',
                     '--config-file', 'vmware-appliance.yaml')
                 if err:
-                    if not confirm_prompt('Deployment failed. Still want to proceed?'):
+                    # Removing confirm_prompts for automated testing
+                    # Considering if deploy command fails, we fail the automation
+                    if self._isAutomated or not confirm_prompt('Deployment failed. Still want to proceed?'):
```

Comment:
Shouldn't we also print reason for deployment failure instead of just printing deployment failed? Might be trimmed version of error so that user is aware about failed.

Reply:
The error is already logged by az cli infra https://github.com/Azure/ArcOnAVS/blob/db6d3fe737b12380114c7a2d0cc69718177937c7/src/appliance-onboarding-script/appliance_setup/pkgs/_az_cli.py#L20

Reply:
As mentioned, we log results in az cli module.